virttest/virt_vm.py

```diff
@@ -644,6 +644,28 @@ def get_mac_address(self, nic_index=0):
 
     def get_address(self, index=0, ip_version="ipv4"):
         """
+        Wrapper for self._get_address. if 'flexible_nic_index' is 'yes',
+        will traverses from first nic to first element toward the end
+        util get a reachable IP address;
+        """
+        flexible_nic_index = ("yes" == self.params.get("flexible_nic_index", "no"))
+        macs, nics_index = self.virtnet.mac_list(), [index]
+        if flexible_nic_index:
+            nics_index += [i for i in xrange(len(macs)) if i != index]
+        flexible_nic_index &= bool(len(macs) > 1)
+        for index in nics_index:
+            try:
+                return self._get_address(index, ip_version)
+            except VMIPAddressMissingError:
+                if not flexible_nic_index:
+                    raise
+                else:
+                    continue
+        else:
+            raise
```

Comment:
But the "real" exception is a little misleading. If we get here, it means no reachable IP addrs have been found for all the NICs, not just for the last one.